MovieQuiz/Services/QuestionFactory.swift

```diff
@@ -51,7 +55,10 @@ class QuestionFactory: QuestionFactoryProtocol {
             do {
                 imageData = try Data(contentsOf: movie.resizedImageUrl)
             } catch {
-                print("Failed to load image")
+                DispatchQueue.main.async { [weak self] in
+                    guard let self = self else { return }
+                    self.delegate?.didFailToLoadData(with: QuestionError.imageError)
+                }
```

Comment:
тут вот аккуратно надо следить, куда выполнение программы идет дальше. 
В вашем случае - сразу дособирает вопрос и вернет его без картинки в приложение 
```suggestion
                DispatchQueue.main.async { [weak self] in
                    self?.delegate?.didFailToLoadData(with: QuestionError.imageError)
                }
                return
```